Fix #385 Searching on multiple tags is broken

oli-ver opened the pull request:
Added simple exists sql clause to find all experiments or database
entries that contains the tag or both of the chosen tags.

Still problems setting up my development environment. Now finally just one commit...

oli-ver commented:
Pull Request for #385

NicolasCARPi commented on search.php:
```diff
@@ -290,13 +290,15 @@
     // TABLE
     if (isset($_GET['type']) && $_GET['type'] === 'experiments') {
         $table = 'experiments';
+        $tableTag = 'items_tags';
```
Shouldn't this be `experiments_tags`?

oli-ver commented:
Yes, you are right. I must have made a copy and paste error, when I recreated the branch.

oli-ver commented:
1) Changed it
2) Fixed it
3) I believe yes, because it is a subquery. But I tried, just to be sure:

Base table or view not found: 1146 Table 'elabftw.tagt' doesn't exist in /elabftw/app/models/Entity.php:172

4) Can you please be more specific? I am not sure that I understand. Where would you implement this left join? Here is my generated SQL, if you query for two tags:

```sql
SELECT DISTINCT experiments.*,
         status.color,
         status.name AS category,
         status.id AS category_id,
         uploads.up_item_id,
         uploads.has_attachment ,
         GROUP_CONCAT(tagt.tag SEPARATOR '|') AS tags, GROUP_CONCAT(tagt.id) AS tags_id , experiments_comments.recentComment
FROM experiments
LEFT JOIN experiments_tags AS tagt
    ON (experiments.id = tagt.item_id)
LEFT JOIN status
    ON (status.id = experiments.status)
LEFT JOIN 
    (SELECT uploads.item_id AS up_item_id,
         (uploads.item_id IS NOT NULL) AS has_attachment,
         uploads.type
    FROM uploads
    GROUP BY  uploads.item_id, uploads.type) AS uploads
    ON (uploads.up_item_id = experiments.id
        AND uploads.type = 'experiments')
LEFT JOIN 
    (SELECT MAX(experiments_comments.datetime) AS recentComment,
         experiments_comments.exp_id
    FROM experiments_comments
    GROUP BY  experiments_comments.exp_id ) AS experiments_comments
    ON (experiments_comments.exp_id = experiments.id)
WHERE experiments.team = :team
        AND experiments.userid = 1
        AND EXISTS 
    (SELECT 1
    FROM experiments_tags tagt
    WHERE tagt.item_id = experiments.id
            AND tagt.tag LIKE '%A%')
        AND EXISTS 
    (SELECT 1
    FROM experiments_tags tagt
    WHERE tagt.item_id = experiments.id
            AND tagt.tag LIKE '%B%')
GROUP BY  id
ORDER BY  id DESC
```
5) I am really sorry :) this time I tested more thoroughly. I found and fixed another bug in the search. It seems the reset of the tags does not work when switching between the database and experiment search. So after you have set a tag for the experiment and then search in the database you will have http get parameters for both of the tables. The database tags "win" because of this code (seach.php, from line 125):

```php
$tagsArr = array();
if (isset($_GET['tag_exp'])) {
    $tagsArr = $_GET['tag_exp'];
}
if (isset($_GET['tag_db'])) {
    $tagsArr = $_GET['tag_db'];
}

```

I fixed this by checking the type get parameter as well.

NicolasCARPi commented:
Hello,

Thank you for your contribution. Your code has been merged :)

Cheers,
~Nico
